### PR TITLE
[Python] Add UNION_BY_NAME to from_parquet arguments

### PR DESCRIPTION
**Pull request by papparapa**
This PR adds `UNION_BY_NAME` merged in #5716 to arguments of `from_parquet` function in Python client.

### CODE REVIEW

File: tools/pythonpkg/tests/fast/test_parquet.py [hunk 1]
```diff
@@ -110,5 +112,29 @@ def test_from_parquet_binary_as_string_default_conn(self,duckdb_cursor):
         res = rel.execute().fetchall()
         assert res[0] == ('foo',)
 
+    def test_from_parquet_union_by_name(self, duckdb_cursor):
+        conn = duckdb.connect()
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/test_parquet.py:116]: Can you switch this to using the pytest temp path instead?

File: tools/pythonpkg/duckdb-stubs/__init__.pyi [hunk 2]
```diff
@@ -67,9 +67,9 @@ class DuckDBPyConnection:
     def from_csv_auto(self, file_name: str) -> DuckDBPyRelation: ...
     def from_df(self, df: pandas.DataFrame = ...) -> DuckDBPyRelation: ...
     @overload
-    def from_parquet(self, file_glob: str, binary_as_string: bool = ..., *, file_row_number: bool = ..., filename: bool = ..., hive_partitioning: bool = ...) -> DuckDBPyRelation: ...
+    def from_parquet(self, file_glob: str, *, binary_as_string: bool = ..., file_row_number: bool = ..., filename: bool = ..., hive_partitioning: bool = ..., union_by_name: bool = ...) -> DuckDBPyRelation: ...
```
Comment (Mause) [inline tools/pythonpkg/duckdb-stubs/__init__.pyi:70]: Can you move the stars in these signatures back to the original correct location?

File: tools/pythonpkg/duckdb_python.cpp [hunk 3]
```diff
@@ -217,11 +219,13 @@ PYBIND11_MODULE(DUCKDB_PYTHON_LIB_NAME, m) {
...
 	      "Creates a relation object from the Parquet files in file_glob", py::arg("file_glob"),
 	      py::arg("binary_as_string") = false, py::kw_only(), py::arg("file_row_number") = false,
-	      py::arg("filename") = false, py::arg("hive_partitioning") = false, py::arg("connection") = py::none());
+	      py::arg("filename") = false, py::arg("hive_partitioning") = false, py::arg("union_by_name"),
+	      py::arg("connection") = py::none());
 	m.def("from_parquet", &DuckDBPyRelation::FromParquets,
 	      "Creates a relation object from the Parquet files in file_globs", py::arg("file_globs"),
 	      py::arg("binary_as_string") = false, py::kw_only(), py::arg("file_row_number") = false,
-	      py::arg("filename") = false, py::arg("hive_partitioning") = false, py::arg("connection") = py::none());
+	      py::arg("filename") = false, py::arg("hive_partitioning") = false, py::arg("union_by_name"),
```
Comment (Mause) [inline tools/pythonpkg/duckdb_python.cpp:227]: You haven't provided a default here - does pybind11 insert one automatically?

**Review by Mause — CHANGES_REQUESTED** [review]

File: tools/pythonpkg/duckdb-stubs/__init__.pyi (hunk 2 above)
Comment (papparapa) [reply tools/pythonpkg/duckdb-stubs/__init__.pyi]: Fixed

File: tools/pythonpkg/tests/fast/test_parquet.py (hunk 1 above)
Comment (papparapa) [reply tools/pythonpkg/tests/fast/test_parquet.py:122]: Fixed

**Comment by Mytherin** [comment]
Thanks for the PR! LGTM

**Pull request merged**